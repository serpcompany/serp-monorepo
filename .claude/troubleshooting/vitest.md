https://vitest.dev/guide/common-errors.html