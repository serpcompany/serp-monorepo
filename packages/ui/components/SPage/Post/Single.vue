<template>
  <div>
    <main>
      <SinglePostsGlossaryPost v-if="data.module === 'Glossary'" :data="data" />
      <SinglePostsPost v-else :data="data" />
    </main>
  </div>
</template>

<script setup lang="ts">
const props = defineProps<{ module?: string }>();

const computedModule = computed(() => props.module);

const route = useRoute();
const slug = route.params.slug as string;
const router = useRouter();
const data = await usePost(slug, computedModule.value);
if (!data) {
  router.push('/404');
}

useSeoMeta({
  title: () => data.title
});
</script>
