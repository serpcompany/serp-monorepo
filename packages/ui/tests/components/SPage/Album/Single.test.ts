import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import Single from '../../../../components/SPage/Album/Single.vue';
import ComponentRender from '../../../componentRender';

let useAuth_: boolean = true;
const scenarios: [
  string,
  { album: Record<string, unknown>; upvotes: string[]; useAuth: boolean }
][] = [
  [
    'with auth enabled and full album data',
    {
      album: {
        name: 'Test Album',
        slug: 'test-album',
        overview: 'This is a test album overview.',
        recordings: [
          { slug: 'song-one', name: 'Song One', position: 1, has_lyrics: true },
          { slug: 'song-two', name: 'Song Two', position: 2, has_lyrics: false }
        ],
        coverArt: { '500': '/test-cover.jpg' },
        date: '2020-01-01',
        genres: ['Rock', 'Pop'],
        tags: ['Indie', 'Live'],
        id: 'album123',
        seoTitle: 'Test Album SEO Title',
        seoDescription: 'Test Album SEO Description'
      },
      upvotes: ['user@example.com'],
      useAuth: true
    }
  ],
  [
    'with auth disabled and minimal album data',
    {
      album: {
        name: 'Minimal Album',
        slug: 'minimal-album',
        overview: 'Minimal album overview.',
        recordings: [],
        coverArt: {},
        date: null,
        genres: [],
        tags: [],
        id: 'album456',
        seoTitle: '',
        seoDescription: ''
      },
      upvotes: [],
      useAuth: false
    }
  ],
  [
    'with auth enabled and no recordings',
    {
      album: {
        name: 'No Songs Album',
        slug: 'no-songs-album',
        overview: 'Overview of album with no songs.',
        recordings: [],
        coverArt: { '500': '/no-songs-cover.jpg' },
        date: '2021-06-15',
        genres: ['Jazz'],
        tags: ['Smooth'],
        id: 'album789',
        seoTitle: 'No Songs SEO',
        seoDescription: 'No Songs Description'
      },
      upvotes: ['anotheruser@example.com'],
      useAuth: true
    }
  ]
];

describe('SPageAlbumSingle Snapshot', () => {
  it.each(scenarios)(
    'renders %s correctly',
    async (desc, { album, upvotes, useAuth }) => {
      globalThis.useAlbum = () => Promise.resolve(album);
      globalThis.useFetchWithCache = () => Promise.resolve({ upvotes });
      useAuth_ = useAuth;
      mockNuxtImport('useRuntimeConfig', () => () => ({
        app: { baseURL: '/' },
        public: { useAuth_ }
      }));
      mockNuxtImport('useSeoMeta', () => () => {});

      const html = await ComponentRender(
        `SPageAlbumSingle ${desc}`,
        {},
        Single
      );
      expect(html).toMatchSnapshot();
    }
  );
});
