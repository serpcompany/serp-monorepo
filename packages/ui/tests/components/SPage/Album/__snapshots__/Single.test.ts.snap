// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageAlbumSingle Snapshot > renders with auth disabled and minimal album data correctly 1`] = `
<div>
  <header class="
    bg-background
    bg-background
    border-b
    border-gray-200
    dark:border-gray-800
    duration-300
    duration-300
    sticky
    sticky
    top-0
    top-0
    transition-all
    transition-all
    z-10
    z-40
  ">
    <div class="
      lg:px-8
      max-w-(--ui-container)
      mx-auto
      px-4
      sm:px-6
      w-full
    ">
      <div class="
        flex
        flex-col
        gap-4
        items-center
        justify-between
        py-3
        sm:flex-row
      ">
        <div class="
          flex
          flex-grow
          gap-3
          items-center
          min-w-0
        ">
          <!-- v-if -->
          <div class="
            flex-grow
            min-w-0
          ">
            <h1 class="
              dark:text-white
              duration-300
              font-bold
              sm:text-2xl
              text-gray-900
              text-xl
              transition-all
              truncate
            ">
              Minimal Album
            </h1>
            <div class="
              dark:text-gray-400
              text-gray-500
              text-sm
              truncate
            "></div>
          </div>
        </div>
        <div class="
          flex
          flex-shrink-0
          gap-3
          items-center
          justify-end
          sm:w-auto
          w-full
        ">
          <a
            class="
              aria-disabled:cursor-not-allowed
              aria-disabled:opacity-75
              disabled:cursor-not-allowed
              disabled:opacity-75
              font-medium
              gap-1.5
              inline-flex
              items-center
              px-2.5
              py-1.5
              rounded-md
              sm:w-auto
              text-xs
              transition-colors
              w-full
            "
            href="https://serp.ly/@daftfm/amazon/music/unlimited"
            rel="noopener noreferrer"
            target="_blank"
          ><span
            aria-hidden="true"
            class="
              i-heroicons:arrow-top-right-on-square
              iconify
              shrink-0
              size-4
            "
          ></span><span class="truncate">Listen Free</span>
            
          <!-- v-if -->
          </a>
        </div>
      </div>
      <nav class="
        -mx-4
        border-gray-200
        border-t
        dark:border-gray-800
        flex
        overflow-x-auto
        px-4
        sm:-mx-6
        sm:px-6
      ">
        <a
          class="
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            disabled:cursor-not-allowed
            disabled:opacity-75
            flex-shrink-0
            font-medium
            gap-1.5
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            text-xs
            transition-colors
            whitespace-nowrap
          "
          href="#overview"
        >
          
        <!-- v-if --><span class="truncate">Overview</span>
          
        <!-- v-if -->
        </a>
        <a
          class="
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            disabled:cursor-not-allowed
            disabled:opacity-75
            flex-shrink-0
            font-medium
            gap-1.5
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            text-xs
            transition-colors
            whitespace-nowrap
          "
          href="#songs"
        >
          
        <!-- v-if --><span class="truncate">Songs</span>
          
        <!-- v-if -->
        </a>
      </nav>
    </div>
  </header>
  <!-- Main content wrapper -->
  <div class="
    lg:p-8
    max-w-7xl
    md:p-6
    mx-auto
    p-4
  ">
    <!-- Breadcrumbs -->
    <nav
      aria-label="breadcrumb"
      class="
        mb-6
        min-w-0
        relative
      "
    >
      <ol class="
        flex
        gap-1.5
        items-center
      "></ol>
    </nav>
    <!-- Grid Layout -->
    <div class="
      gap-6
      grid
      lg:grid-cols-3
    ">
      <!-- Main Content Column (70%) -->
      <div class="
        lg:col-span-2
        space-y-8
      ">
        <!-- Songs / Tracklist Section -->
        <section
          class="scroll-mt-24"
          data-adsense-content="album-tracklist"
          id="songs"
        >
          <h2 class="
            font-semibold
            mb-4
            text-2xl
          ">
            Tracklist
          </h2>
          <div class="
            bg-default
            divide-default
            divide-y
            ring
            ring-default
            rounded-lg
          ">
            <!-- v-if -->
            <div class="
              Object]
              [object
              p-4
              sm:p-6
            ">
              <p class="
                dark:text-gray-400
                text-gray-500
                text-sm
              ">
                No tracklist available.
              </p>
            </div>
            <!-- v-if -->
          </div>
        </section>
        <!-- Overview Section -->
        <section
          class="scroll-mt-24"
          data-adsense-content="album-overview"
          id="overview"
        >
          <h2 class="
            font-semibold
            mb-4
            text-2xl
          ">
            Overview
          </h2>
          <div class="
            dark:prose-invert
            max-w-none
            prose
          ">
            Minimal album overview.
          </div>
        </section>
        <!-- Other Albums by Artist Section -->
        <!-- v-if -->
      </div>
      <!-- Sidebar Column (30%) -->
      <div class="
        lg:col-span-1
        space-y-6
      ">
        <!-- Album image -->
        <!-- v-if -->
        <!-- Album Info Card -->
        <div class="
          bg-default
          divide-default
          divide-y
          ring
          ring-default
          rounded-lg
        ">
          <div class="
            p-4
            sm:px-6
          ">
            <div class="font-medium">
              Album Info
            </div>
          </div>
          <div class="
            p-4
            sm:p-6
          ">
            <div class="
              space-y-2
              text-sm
            ">
              <!-- Artist Link -->
              <!-- v-if -->
              <!-- v-if -->
              <!-- v-if -->
            </div>
          </div>
          <!-- v-if -->
        </div>
        <!-- Stats Card -->
        <!-- v-if -->
        <!-- Links Card -->
        <div class="
          bg-default
          divide-default
          divide-y
          ring
          ring-default
          rounded-lg
        ">
          <div class="
            p-4
            sm:px-6
          ">
            <div class="font-medium">
              Listen On
            </div>
          </div>
          <div class="
            p-4
            sm:p-6
          ">
            <div class="p-1">
              <a
                class="
                  aria-disabled:bg-primary/10
                  aria-disabled:cursor-not-allowed
                  aria-disabled:opacity-75
                  bg-blue-50
                  dark:bg-blue-900/20
                  dark:hover:bg-blue-900/30
                  disabled:bg-primary/10
                  disabled:cursor-not-allowed
                  disabled:opacity-75
                  focus-visible:bg-primary/15
                  focus:outline-none
                  font-medium
                  gap-2
                  hover:bg-blue-100
                  inline-flex
                  items-center
                  justify-center
                  px-3
                  py-2
                  rounded-md
                  text-primary
                  text-sm
                  transition-colors
                  w-full
                "
                href="https://serp.ly/@daftfm/amazon/music/unlimited"
                rel="noopener noreferrer"
              >
                
              <!-- v-if --> Amazon Music
                
              <!-- v-if -->
              </a>
            </div>
          </div>
          <!-- v-if -->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SPageAlbumSingle Snapshot > renders with auth enabled and full album data correctly 1`] = `
<div>
  <header class="
    bg-background
    bg-background
    border-b
    border-gray-200
    dark:border-gray-800
    duration-300
    duration-300
    sticky
    sticky
    top-0
    top-0
    transition-all
    transition-all
    z-10
    z-40
  ">
    <div class="
      lg:px-8
      max-w-(--ui-container)
      mx-auto
      px-4
      sm:px-6
      w-full
    ">
      <div class="
        flex
        flex-col
        gap-4
        items-center
        justify-between
        py-3
        sm:flex-row
      ">
        <div class="
          flex
          flex-grow
          gap-3
          items-center
          min-w-0
        ">
          <!-- v-if -->
          <div class="
            flex-grow
            min-w-0
          ">
            <h1 class="
              dark:text-white
              duration-300
              font-bold
              sm:text-2xl
              text-gray-900
              text-xl
              transition-all
              truncate
            ">
              Test Album
            </h1>
            <div class="
              dark:text-gray-400
              text-gray-500
              text-sm
              truncate
            "></div>
          </div>
        </div>
        <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto">
          <div class="relative inline-flex -space-x-px" module="music_albums"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-5" aria-hidden="true"></span>
              <!--v-if-->
              <!--v-if-->
            </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] gap-1 rounded-md ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center text-sm"><!--v-if--><span class="truncate">NaN</span>
            <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-5" aria-hidden="true"></span>
              <!--v-if-->
              <!--v-if-->
            </button>
          </div><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
            <!--v-if-->
          </a>
        </div>
      </div>
      <nav class="
        -mx-4
        border-gray-200
        border-t
        dark:border-gray-800
        flex
        overflow-x-auto
        px-4
        sm:-mx-6
        sm:px-6
      ">
        <a
          class="
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            disabled:cursor-not-allowed
            disabled:opacity-75
            flex-shrink-0
            font-medium
            gap-1.5
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            text-xs
            transition-colors
            whitespace-nowrap
          "
          href="#overview"
        >
          
        <!-- v-if --><span class="truncate">Overview</span>
          
        <!-- v-if -->
        </a>
        <a
          class="
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            disabled:cursor-not-allowed
            disabled:opacity-75
            flex-shrink-0
            font-medium
            gap-1.5
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            text-xs
            transition-colors
            whitespace-nowrap
          "
          href="#songs"
        >
          
        <!-- v-if --><span class="truncate">Songs</span>
          
        <!-- v-if -->
        </a>
      </nav>
    </div>
  </header>
  <!-- Main content wrapper -->
  <div class="
    lg:p-8
    max-w-7xl
    md:p-6
    mx-auto
    p-4
  ">
    <!-- Breadcrumbs -->
    <nav
      aria-label="breadcrumb"
      class="
        mb-6
        min-w-0
        relative
      "
    >
      <ol class="
        flex
        gap-1.5
        items-center
      "></ol>
    </nav>
    <!-- Grid Layout -->
    <div class="
      gap-6
      grid
      lg:grid-cols-3
    ">
      <!-- Main Content Column (70%) -->
      <div class="
        lg:col-span-2
        space-y-8
      ">
        <!-- Songs / Tracklist Section -->
        <section
          class="scroll-mt-24"
          data-adsense-content="album-tracklist"
          id="songs"
        >
          <h2 class="
            font-semibold
            mb-4
            text-2xl
          ">
            Tracklist
          </h2>
          <div class="
            bg-default
            divide-default
            divide-y
            ring
            ring-default
            rounded-lg
          ">
            <!-- v-if -->
            <div class="
              Object]
              [object
              p-4
              sm:p-6
            ">
              <ol class="space-y-2">
                <li class="
                  border-b
                  border-gray-200
                  dark:border-gray-700
                  flex
                  gap-2
                  items-center
                  justify-between
                  last:border-b-0
                  last:pb-0
                  pb-2
                ">
                  <div class="
                    flex
                    gap-3
                    items-center
                    min-w-0
                  ">
                    <span class="
                      dark:text-gray-400
                      font-mono
                      text-gray-500
                      text-right
                      text-xs
                      w-5
                    ">
                      1.
                    </span>
                    <div class="truncate">
                      <a
                        class="
                          dark:text-primary-400
                          font-medium
                          hover:underline
                          text-primary-600
                          text-sm
                          truncate
                        "
                        href="/songs/song-one"
                        title="Song One"
                      >Song One</a>
                    </div>
                  </div>
                  <span class="
                    dark:text-gray-400
                    flex-shrink-0
                    font-mono
                    text-gray-500
                    text-xs
                  ">
                    --:--
                  </span>
                </li>
                <li class="
                  border-b
                  border-gray-200
                  dark:border-gray-700
                  flex
                  gap-2
                  items-center
                  justify-between
                  last:border-b-0
                  last:pb-0
                  pb-2
                ">
                  <div class="
                    flex
                    gap-3
                    items-center
                    min-w-0
                  ">
                    <span class="
                      dark:text-gray-400
                      font-mono
                      text-gray-500
                      text-right
                      text-xs
                      w-5
                    ">
                      2.
                    </span>
                    <div class="truncate">
                      <span
                        class="
                          dark:text-gray-200
                          text-gray-800
                          text-sm
                          truncate
                        "
                        title="Song Two"
                      >
                        Song Two
                      </span>
                    </div>
                  </div>
                  <span class="
                    dark:text-gray-400
                    flex-shrink-0
                    font-mono
                    text-gray-500
                    text-xs
                  ">
                    --:--
                  </span>
                </li>
              </ol>
            </div>
            <!-- v-if -->
          </div>
        </section>
        <!-- Overview Section -->
        <section
          class="scroll-mt-24"
          data-adsense-content="album-overview"
          id="overview"
        >
          <h2 class="
            font-semibold
            mb-4
            text-2xl
          ">
            Overview
          </h2>
          <div class="
            dark:prose-invert
            max-w-none
            prose
          ">
            This is a test album overview.
          </div>
        </section>
        <!-- Other Albums by Artist Section -->
        <!-- v-if -->
      </div>
      <!-- Sidebar Column (30%) -->
      <div class="
        lg:col-span-1
        space-y-6
      ">
        <!-- Album image -->
        <div>
          <img
            alt="Test Album cover art"
            class="
              object-cover
              rounded-md
              shadow-md
              w-full
            "
            data-nuxt-img=""
            src="/_ipx/_/test-cover.jpg"
            srcset="/_ipx/_/test-cover.jpg 1x, /_ipx/_/test-cover.jpg 2x"
          />
        </div>
        <!-- Album Info Card -->
        <div class="
          bg-default
          divide-default
          divide-y
          ring
          ring-default
          rounded-lg
        ">
          <div class="
            p-4
            sm:px-6
          ">
            <div class="font-medium">
              Album Info
            </div>
          </div>
          <div class="
            p-4
            sm:p-6
          ">
            <div class="
              space-y-2
              text-sm
            ">
              <!-- Artist Link -->
              <!-- v-if -->
              <div>
                <span class="
                  dark:text-gray-300
                  font-medium
                  text-gray-700
                ">
                  Released:
                </span>
                <span class="
                  dark:text-gray-200
                  text-gray-800
                ">
                  2020
                </span>
              </div>
              <!-- v-if -->
            </div>
          </div>
          <!-- v-if -->
        </div>
        <!-- Stats Card -->
        <div class="
          bg-default
          divide-default
          divide-y
          ring
          ring-default
          rounded-lg
        ">
          <div class="
            p-4
            sm:px-6
          ">
            <div class="font-medium">
              Stats
            </div>
          </div>
          <div class="
            p-4
            sm:p-6
          ">
            <div class="
              space-y-2
              text-sm
            ">
              <div>
                <span class="
                  dark:text-gray-300
                  font-medium
                  text-gray-700
                ">
                  Genres:
                </span>
                <span class="
                  dark:text-gray-200
                  text-gray-800
                ">
                  Rock, Pop
                </span>
              </div>
              <div>
                <span class="
                  dark:text-gray-300
                  font-medium
                  text-gray-700
                ">
                  Tags:
                </span>
                <span class="
                  dark:text-gray-200
                  text-gray-800
                ">
                  Indie, Live
                </span>
              </div>
            </div>
          </div>
          <!-- v-if -->
        </div>
        <!-- Links Card -->
        <div class="
          bg-default
          divide-default
          divide-y
          ring
          ring-default
          rounded-lg
        ">
          <div class="
            p-4
            sm:px-6
          ">
            <div class="font-medium">
              Listen On
            </div>
          </div>
          <div class="
            p-4
            sm:p-6
          ">
            <div class="p-1">
              <a
                class="
                  aria-disabled:bg-primary/10
                  aria-disabled:cursor-not-allowed
                  aria-disabled:opacity-75
                  bg-blue-50
                  dark:bg-blue-900/20
                  dark:hover:bg-blue-900/30
                  disabled:bg-primary/10
                  disabled:cursor-not-allowed
                  disabled:opacity-75
                  focus-visible:bg-primary/15
                  focus:outline-none
                  font-medium
                  gap-2
                  hover:bg-blue-100
                  inline-flex
                  items-center
                  justify-center
                  px-3
                  py-2
                  rounded-md
                  text-primary
                  text-sm
                  transition-colors
                  w-full
                "
                href="https://serp.ly/@daftfm/amazon/music/unlimited"
                rel="noopener noreferrer"
              >
                
              <!-- v-if --> Amazon Music
                
              <!-- v-if -->
              </a>
            </div>
          </div>
          <!-- v-if -->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SPageAlbumSingle Snapshot > renders with auth enabled and no recordings correctly 1`] = `
<div>
  <header class="
    bg-background
    bg-background
    border-b
    border-gray-200
    dark:border-gray-800
    duration-300
    duration-300
    sticky
    sticky
    top-0
    top-0
    transition-all
    transition-all
    z-10
    z-40
  ">
    <div class="
      lg:px-8
      max-w-(--ui-container)
      mx-auto
      px-4
      sm:px-6
      w-full
    ">
      <div class="
        flex
        flex-col
        gap-4
        items-center
        justify-between
        py-3
        sm:flex-row
      ">
        <div class="
          flex
          flex-grow
          gap-3
          items-center
          min-w-0
        ">
          <!-- v-if -->
          <div class="
            flex-grow
            min-w-0
          ">
            <h1 class="
              dark:text-white
              duration-300
              font-bold
              sm:text-2xl
              text-gray-900
              text-xl
              transition-all
              truncate
            ">
              No Songs Album
            </h1>
            <div class="
              dark:text-gray-400
              text-gray-500
              text-sm
              truncate
            "></div>
          </div>
        </div>
        <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto">
          <div class="relative inline-flex -space-x-px" module="music_albums"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-5" aria-hidden="true"></span>
              <!--v-if-->
              <!--v-if-->
            </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] gap-1 rounded-md ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center text-sm"><!--v-if--><span class="truncate">NaN</span>
            <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-5" aria-hidden="true"></span>
              <!--v-if-->
              <!--v-if-->
            </button>
          </div><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
            <!--v-if-->
          </a>
        </div>
      </div>
      <nav class="
        -mx-4
        border-gray-200
        border-t
        dark:border-gray-800
        flex
        overflow-x-auto
        px-4
        sm:-mx-6
        sm:px-6
      ">
        <a
          class="
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            disabled:cursor-not-allowed
            disabled:opacity-75
            flex-shrink-0
            font-medium
            gap-1.5
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            text-xs
            transition-colors
            whitespace-nowrap
          "
          href="#overview"
        >
          
        <!-- v-if --><span class="truncate">Overview</span>
          
        <!-- v-if -->
        </a>
        <a
          class="
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            disabled:cursor-not-allowed
            disabled:opacity-75
            flex-shrink-0
            font-medium
            gap-1.5
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            text-xs
            transition-colors
            whitespace-nowrap
          "
          href="#songs"
        >
          
        <!-- v-if --><span class="truncate">Songs</span>
          
        <!-- v-if -->
        </a>
      </nav>
    </div>
  </header>
  <!-- Main content wrapper -->
  <div class="
    lg:p-8
    max-w-7xl
    md:p-6
    mx-auto
    p-4
  ">
    <!-- Breadcrumbs -->
    <nav
      aria-label="breadcrumb"
      class="
        mb-6
        min-w-0
        relative
      "
    >
      <ol class="
        flex
        gap-1.5
        items-center
      "></ol>
    </nav>
    <!-- Grid Layout -->
    <div class="
      gap-6
      grid
      lg:grid-cols-3
    ">
      <!-- Main Content Column (70%) -->
      <div class="
        lg:col-span-2
        space-y-8
      ">
        <!-- Songs / Tracklist Section -->
        <section
          class="scroll-mt-24"
          data-adsense-content="album-tracklist"
          id="songs"
        >
          <h2 class="
            font-semibold
            mb-4
            text-2xl
          ">
            Tracklist
          </h2>
          <div class="
            bg-default
            divide-default
            divide-y
            ring
            ring-default
            rounded-lg
          ">
            <!-- v-if -->
            <div class="
              Object]
              [object
              p-4
              sm:p-6
            ">
              <p class="
                dark:text-gray-400
                text-gray-500
                text-sm
              ">
                No tracklist available.
              </p>
            </div>
            <!-- v-if -->
          </div>
        </section>
        <!-- Overview Section -->
        <section
          class="scroll-mt-24"
          data-adsense-content="album-overview"
          id="overview"
        >
          <h2 class="
            font-semibold
            mb-4
            text-2xl
          ">
            Overview
          </h2>
          <div class="
            dark:prose-invert
            max-w-none
            prose
          ">
            Overview of album with no songs.
          </div>
        </section>
        <!-- Other Albums by Artist Section -->
        <!-- v-if -->
      </div>
      <!-- Sidebar Column (30%) -->
      <div class="
        lg:col-span-1
        space-y-6
      ">
        <!-- Album image -->
        <div>
          <img
            alt="No Songs Album cover art"
            class="
              object-cover
              rounded-md
              shadow-md
              w-full
            "
            data-nuxt-img=""
            src="/_ipx/_/no-songs-cover.jpg"
            srcset="/_ipx/_/no-songs-cover.jpg 1x, /_ipx/_/no-songs-cover.jpg 2x"
          />
        </div>
        <!-- Album Info Card -->
        <div class="
          bg-default
          divide-default
          divide-y
          ring
          ring-default
          rounded-lg
        ">
          <div class="
            p-4
            sm:px-6
          ">
            <div class="font-medium">
              Album Info
            </div>
          </div>
          <div class="
            p-4
            sm:p-6
          ">
            <div class="
              space-y-2
              text-sm
            ">
              <!-- Artist Link -->
              <!-- v-if -->
              <div>
                <span class="
                  dark:text-gray-300
                  font-medium
                  text-gray-700
                ">
                  Released:
                </span>
                <span class="
                  dark:text-gray-200
                  text-gray-800
                ">
                  2021
                </span>
              </div>
              <!-- v-if -->
            </div>
          </div>
          <!-- v-if -->
        </div>
        <!-- Stats Card -->
        <div class="
          bg-default
          divide-default
          divide-y
          ring
          ring-default
          rounded-lg
        ">
          <div class="
            p-4
            sm:px-6
          ">
            <div class="font-medium">
              Stats
            </div>
          </div>
          <div class="
            p-4
            sm:p-6
          ">
            <div class="
              space-y-2
              text-sm
            ">
              <div>
                <span class="
                  dark:text-gray-300
                  font-medium
                  text-gray-700
                ">
                  Genres:
                </span>
                <span class="
                  dark:text-gray-200
                  text-gray-800
                ">
                  Jazz
                </span>
              </div>
              <div>
                <span class="
                  dark:text-gray-300
                  font-medium
                  text-gray-700
                ">
                  Tags:
                </span>
                <span class="
                  dark:text-gray-200
                  text-gray-800
                ">
                  Smooth
                </span>
              </div>
            </div>
          </div>
          <!-- v-if -->
        </div>
        <!-- Links Card -->
        <div class="
          bg-default
          divide-default
          divide-y
          ring
          ring-default
          rounded-lg
        ">
          <div class="
            p-4
            sm:px-6
          ">
            <div class="font-medium">
              Listen On
            </div>
          </div>
          <div class="
            p-4
            sm:p-6
          ">
            <div class="p-1">
              <a
                class="
                  aria-disabled:bg-primary/10
                  aria-disabled:cursor-not-allowed
                  aria-disabled:opacity-75
                  bg-blue-50
                  dark:bg-blue-900/20
                  dark:hover:bg-blue-900/30
                  disabled:bg-primary/10
                  disabled:cursor-not-allowed
                  disabled:opacity-75
                  focus-visible:bg-primary/15
                  focus:outline-none
                  font-medium
                  gap-2
                  hover:bg-blue-100
                  inline-flex
                  items-center
                  justify-center
                  px-3
                  py-2
                  rounded-md
                  text-primary
                  text-sm
                  transition-colors
                  w-full
                "
                href="https://serp.ly/@daftfm/amazon/music/unlimited"
                rel="noopener noreferrer"
              >
                
              <!-- v-if --> Amazon Music
                
              <!-- v-if -->
              </a>
            </div>
          </div>
          <!-- v-if -->
        </div>
      </div>
    </div>
  </div>
</div>
`;
